lib/Frontend/Frontend.cpp

```diff
@@ -283,16 +283,14 @@ bool CompilerInstance::setUpModuleLoaders() {
 }
 
 Optional<unsigned> CompilerInstance::setUpCodeCompletionBuffer() {
-  Optional<unsigned> codeCompletionBufferID;
+  if (!Invocation.isCodeCompletion())
```

Comment:
I like the early return, but it relies on an invariant: if isCodeCompletion is true, then getCodeCompletionPoint will not return null. Is there some other place where that invariant is enforced or checked? Or would it be better to always get the code completion point, and early-return if the gotten value were null?